alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHad.js

```diff
@@ -0,0 +1,89 @@
+/*
+
+There are N rectangular buildings standing along the road next to each other. The K-th building is of size H[K] × 1.
+
+Because a renovation of all of the buildings is planned, we want to cover them with rectangular banners until the renovations are finished. Of course, to cover a building, the banner has to be at least as high as the building. We can cover more than one building with a banner if it is wider than 1.
+
+For example, to cover buildings of heights 3, 1, 4 we could use a banner of size 4×3 (i.e. of height 4 and width 3), marked here in blue:
+
+Buildings of sizes (3 × 1), (1 × 1), (4 × 1), covered with scaffolding of size 4×3
+
+We can order at most two banners and we want to cover all of the buildings. Also, we want to minimize the amount of material needed to produce the banners.
+
+What is the minimum total area of at most two banners which cover all of the buildings?
+
+Write a function:
+
+solution(int[] H)
+
+that, given an array H consisting of N integers, returns the minimum total area of at most two banners that we will have to order.
+
+Examples:
+
+Given H = [3, 1, 4], the function should return 10. The result can be achieved by covering the first two buildings with a banner of size 3×2 and the third building with a banner of size 4×1:
+Illustration of first example
+
+
+Given H = [5, 3, 2, 4], the function should return 17. The result can be achieved by covering the first building with a banner of size 5×1 and the other buildings with a banner of size 4×3:
+Illustration of second example
+Given H = [5, 3, 5, 2, 1], your function should return 19. The result can be achieved by covering the first three buildings with a banner of size 5×3 and the other two with a banner of size 2×2:
+Illustration of third example
+
+
+Given H = [7, 7, 3, 7, 7], your function should return 35. The result can be achieved by using one banner of size 7×5:
+Illustration of fourth example
+
+
+Given H = [1, 1, 7, 6, 6, 6], your function should return 30. The result can be achieved by using banners of size 1×2 and 7×4:
+Illustration of fifth example
+
+
+Write an efficient algorithm for the following assumptions:
+
+N is an integer within the range [1..100,000];
+each element of array H is an integer within the range [1..10,000]
+
+*/
+
+const fs = require("fs");
+const arrBuildings = fs.readFileSync("data.txt", "utf-8").split(",");
+
+function bestArea(arrStreet) {
+    let numberOfBuildings = arrStreet.length;
+    let staHeights = new Array(numberOfBuildings);

```

Comment:
I feel like I only know `sta` is for a stack given our convo thus far.
I'm amused we tried to save 2 characters from this name and have other really long variable names lol
I'd probably just name this `heightStack`

Reply:
Stacks should also start off empty....
but this works perfectly well for a list of `maxHeightAtIndex`. Totally acceptable as well, I'd just probably update the name to match, or keep the same name and actually use it like a stack

Reply:
Why is arr okay, but sta invites scrutiny?  I tried to google if there was a convention, but there didn't seem to be much of one...
I was finding in JS, that if I compared the current value of the stack with a number, and the element of the stack didn't actually exist, it said that NaN was greater than the actual number.  so I decided to prepopulate the stack with naughts to make sure that didn't happen.